package_test.go

```diff
@@ -41,7 +31,30 @@ func TestPackage(t *testing.T) {
 			defer e.RestoreEnv()
 			os.Setenv("PATH", Path)
 
-			WithoutArgs()
+			Convey("Given summon-aws-secrets is run with no arguments", func() {
+				_, stderr, err := RunCommand(PackageName)
+
+				Convey("Returns with error", func() {
+					So(err, ShouldNotBeNil)
+					So(stderr.String(), ShouldStartWith, "A variable ID or version flag must be given as the first and only argument!")
+				})
+			})
+
+			Convey("Given summon-aws-secrets is retrieving a secret with key value format", func() {
```

Comment:
I think it would also be good  to move the tests that you introduced to a seperate function since they're not testing the package but instead are unit tests for the `getValueByKey` function:
```
func TestPackage ...

...

func Test_getValueByKey(t *testing.T) {
	Convey("Given a valid JSON format stored secret", t, func() {
		secret := `{ "username": "USERNAME", "password": "PASSWORD", "port": 8000 }`

		Convey("Returns the value of the key", func() {
			stdout, err := getValueByKey("username", []byte(secret))
			So(err, ShouldBeNil)
			So(string(stdout), ShouldEqual, "USERNAME")
		})

		Convey("Always returns as a string", func() {
			stdout, err := getValueByKey("port", []byte(secret))
			So(err, ShouldBeNil)
			So(string(stdout), ShouldEqual, "8000")
		})
	})
}

```

Reply:
it makes total sense. Thanks for pointing this out.